src/testE2E/amazonq/featureDev.test.ts

```diff
@@ -0,0 +1,135 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as got from 'got'
+import assert from 'assert'
+import { qTestingFramework } from './framework/framework'
+import { FollowUpTypes } from '../../amazonqFeatureDev/types'
+import sinon from 'sinon'
+import { FeatureDevClient } from '../../amazonqFeatureDev/client/featureDev'
+import { verifyTextOrder } from './framework/text'
+import { examples } from '../../amazonqFeatureDev/text'
+import * as authUtil from '../../codewhisperer/util/authUtil'
+
+describe('Amazon Q Feature Dev', function () {
+    let framework: qTestingFramework
+
+    const samplePlanResponse = 'sample plan response'
+
+    beforeEach(() => {
+        /**
+         * TODO remove these stubs when we know the backend can handle all the test load + when we know the tests
+         * are working without any flakiness
+         */
+        sinon.stub(authUtil, 'getChatAuthState').resolves({
+            amazonQ: 'connected',
+            codewhispererChat: 'connected',
+            codewhispererCore: 'connected',
+        })
+        sinon.stub(FeatureDevClient.prototype, 'createConversation').resolves('1234')
+        sinon.stub(FeatureDevClient.prototype, 'createUploadUrl').resolves({
+            uploadId: '5678',
+            uploadUrl: 'foo',
+            $response: sinon.mock() as any,
+        })
+        sinon.stub(FeatureDevClient.prototype, 'generatePlan').resolves(samplePlanResponse)
+        sinon.stub(got, 'default').resolves({ statusCode: 200 })
+        framework = new qTestingFramework('featuredev', true, true)
+    })
+
+    afterEach(() => {
+        framework.dispose()
+        sinon.restore()
+    })
+
+    describe('quick action availability', () => {
+        it('Shows /dev when feature dev is enabled', () => {
+            const q = framework.createTab()
+            const command = q.findCommand('/dev')
+            if (!command) {
+                assert.fail('Could not find command')
+            }
+
+            if (command.length > 1) {
+                assert.fail('Found too many commands with the name /dev')
+            }
+        })
+
+        it('Does NOT show /dev when feature dev is NOT enabled', () => {
+            framework = new qTestingFramework('featuredev', false, true)
+            const q = framework.createTab()
+            const command = q.findCommand('/dev')
+            if (command.length > 0) {
+                assert.fail('Found command when it should not have been found')
+            }
+        })
+    })
+
+    describe('/dev {msg} entry', async () => {
+        it('Receives chat response', async () => {
```

Comment:
question: is this a duplicate from the test below?

Reply:
Yes and no, this is a different entry point. Its when you are writing in the chat `/dev implement fibonacci`. The other one is when you first do `/dev` and then the new tab opens and then you do `implement fibonacci`